sql/core/src/test/scala/org/apache/spark/sql/SQLInsertTestSuite.scala

```diff
@@ -286,20 +286,28 @@ trait SQLInsertTestSuite extends QueryTest with SQLTestUtils {
     } else {
       SQLConf.StoreAssignmentPolicy.values
     }
+
+    def shouldThrowException(policy: SQLConf.StoreAssignmentPolicy.Value): Boolean = policy match {
+      case SQLConf.StoreAssignmentPolicy.ANSI | SQLConf.StoreAssignmentPolicy.STRICT =>
+        true
+      case SQLConf.StoreAssignmentPolicy.LEGACY =>
+        SQLConf.get.ansiEnabled
```

Comment:
one argument here is shall we use non-ANSI cast for legacy store assignment. The legacy config is mostly for restoring the legacy behavior and I don't think ANSI mode should change it.

Reply:
@cloud-fan Since the current code doesn't forbid turning ANSI mode on with legacy store assignment, I am changing the test case in this way.

Reply:
Why don't we change the current code? I believe it's a mistake that we don't forbid it. We forbid users to turn off ansi mode with ansi store assignment, isn't it?

Reply:
@cloud-fan OK, created https://github.com/apache/spark/pull/35539 for this.